mail/utils.py

```
@@ -0,0 +1,82 @@
+        )
+        message = FINANCIAL_AID_APPROVAL_MESSAGE.format(
+            program_name=financial_aid.tier_program.program.title,
+            price=get_formatted_course_price(program_enrollment)["price"]
+        )
+        subject = FINANCIAL_AID_APPROVAL_SUBJECT.format(program_name=financial_aid.tier_program.program.title)
+    elif financial_aid.status == FinancialAidStatus.PENDING_MANUAL_APPROVAL:
+        message = FINANCIAL_AID_DOCUMENTS_RECEIVED_MESSAGE
+        subject = FINANCIAL_AID_DOCUMENTS_RECEIVED_SUBJECT.format(
+            program_name=financial_aid.tier_program.program.title
+        )
+    else:
+        # django.core.exceptions.ValidationError
+        raise ValidationError("Invalid status on FinancialAid for generate_financial_aid_email()")
+    body = FINANCIAL_AID_EMAIL_BODY.format(
+        first_name=financial_aid.user.profile.first_name,
+        message=message,
+        program_name=financial_aid.tier_program.program.title
+    )
+    return {"subject": subject, "body": body}
+
+
+def generate_mailgun_response_json(response):
+    """
+    Generates the json object for the mailgun response.
+
+    This is necessary because of inconsistent Response object formatting. Calling response.json() will return a valid
+    JSON-serializable dictionary object, except when the response returns 401 (and maybe others) from mailgun, in which
+    it will raise an exception because of improperly formatted content for the .json() call.
+
+    This function solves that problem by raising ImproperlyConfigured if the response returns 401, which will be caught
+    by a micromasters.utils.custom_exception_handler():
+
+    Args:
+        response (requests.Response): response object
+    Returns:
+        dict
+    """
+    if response.status_code == status.HTTP_401_UNAUTHORIZED:
+        message = "Mailgun API keys not properly configured."
+        log.error(message)
+        raise ImproperlyConfigured(message)
```

Maybe we are going in too deep details, but given that what I was proposing included dealing with the status code and propagating back the error, I think the 401 is a just a special case where you raise instead of returning.
Anyway, I can accept this solution if you return a proper error code with the json and not just return a 200 in `views.py`